### PR TITLE
Refactor `appveyor.psm1` to `ci.psm1`

### PR DESCRIPTION
**Pull request by RDIL**
## PR Summary
AppVeyor removed, so file has been renamed, as well as many AppVeyor references / env vars / etc. have been removed. 
  

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [X] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [X] Issue filed:  #8686 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [X] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/ci.psm1 [hunk 1]
```diff
@@ -274,7 +221,7 @@ function Invoke-AppVeyorInstall
         1..(Get-Random -Minimum 15 -Maximum 126) | ForEach-Object { $password = $password + [char]$randomObj.next(45,126) }
 
         # Account
-        $userName = 'appVeyorRemote'
+        $userName = 'Remote'
```
Comment (TravisEz13) [inline tools/ci.psm1]: can we give a more descriptive name
```suggestion
        $userName = 'ciRemote'
```

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
Thanks a lot.

Just one comment.

**Comment by TravisEz13** [comment]
This was work we planned.  FYI, We also planned on merging travis.ps1 and removing duplicate code (It doesn't need to be done in the same PR).

File: tools/ci.psm1 [hunk 2]
```diff
@@ -177,27 +168,12 @@ function Invoke-AppVeyorFull
...
         Invoke-AppVeyorTest -ErrorAction Continue
         Invoke-AppveyorFinish
-    }
-    finally {
-        if($APPVEYOR_SCHEDULED_BUILD -and $env:APPVEYOR_SCHEDULED_BUILD)
-        {
-            Remove-Item env:APPVEYOR_SCHEDULED_BUILD
-        }
-    }
+    } finally {}
```
Comment (PaulHigin) [inline tools/ci.psm1]: Since the finally block is empty, we can just remove the try/finally.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: tools/ci.psm1 (hunk 2 above)
Comment (RDIL) [reply tools/ci.psm1]: Done.

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PaulHigin 	Please update your review.

**Comment by RDIL** [comment]
@PaulHigin I fixed the problems, please re-review

**Review by TravisEz13 — APPROVED** [review]

**Comment by RDIL** [comment]
CLA bot is 🅱️roken

**Comment by TravisEz13** [comment]
No worries, The clabot previously approved this PR.

**Comment by TravisEz13** [comment]
We found a test regressed due to an environmental issue.

**Review by PaulHigin — APPROVED** [review]

**Comment by TravisEz13** [comment]
Used admin rights to merge.  CLA did not run on the last commit, but it approved the commit before that.

**Comment by TravisEz13** [comment]
Thanks again @RDIL

**Pull request merged**